src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -427,6 +431,66 @@ private DynamicTraceEventData ReadEventParametersAndBuildTemplate(EventPipeEvent
             return template;
         }
 
+        // The NetPerf and NetTrace V1 file formats were incapable of representing some event parameter types that EventSource and ETW support.
+        // This works around that issue without requiring a runtime or format update for well-known EventSources that used the indescribable types.
+        // Going forward the goal is to update the runtime and file format so these special cases aren't needed, but for backwards compat this is
+        // all we can do.
+        private DynamicTraceEventData.PayloadFetchClassInfo CheckForWellKnownEventFields(EventPipeEventMetaDataHeader eventMetaDataHeader)
```

Comment:
We should only be calling this for existing versions of the file, so we're going to need another change here to make sure that we don't invoke this code once you make the runtime change.